framework/src/modules/chain/transport/broadcaster.js

```diff
@@ -71,189 +58,75 @@ class Broadcaster {
 	}
 
 	/**
-	 * broadcast to network.
-	 *
-	 * @param {Object} params
-	 * @param {Object} options
-	 * @returns {Promise}
-	 * @throws {Error}
-	 * @todo Add description for the params
+	 * Enqueue transactionId into queue if not exists
+	 * @param {string} transactionId  transaction id as string
 	 */
-	async broadcast(params, { api: event, data }) {
-		// Broadcast using Elements P2P library via network module
-		if (event === 'postTransactionsAnnouncement') {
-			return this.channel.invoke('network:broadcast', {
-				event,
-				data,
-			});
+	enqueueTransactionId(transactionId) {
+		if (
+			this.transactionIdQueue.find(id => id === transactionId) !== undefined
```

Comment:
Why not a simple object with transaction ids as keys? Same for signatures. No need for `.find` then, just a check.

Reply:
id queue is `FIFO` queue, so we cannot use map to keep the order